pkg/source/gcs.go

```diff
@@ -13,11 +13,12 @@ var _ Source = (*gcsSource)(nil)
 
 type (
 	GCSConfig struct {
-		Endpoint        string `yaml:"endpoint,omitempty"`
-		CredentialsFile string `yaml:"credentialsFile,omitempty"`
-		CredentialsJSON string `yaml:"credentialsJSON,omitempty"`
-		Bucket          string `yaml:"bucket,omitempty"`
-		Key             string `yaml:"key,omitempty"`
+		Endpoint              string `yaml:"endpoint,omitempty"`
+		CredentialsFile       string `yaml:"credentialsFile,omitempty"`
+		CredentialsJSON       string `yaml:"credentialsJSON,omitempty"`
+		WithoutAuthentication bool   `yaml:"withoutAuthentication,omitempty"`
```

Comment:
The current implementation is that if `CredentialsFile` and `CredentialsJSON` are not configured, then it is considered that there is no authentication. Is there any problem with this?